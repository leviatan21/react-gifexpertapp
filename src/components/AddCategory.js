import React, { useState } from 'react';
import PropTypes from 'prop-types'
//	rafcp
export const AddCategory = ( { setCategories } ) => {
	//
	const initialState = "";
	const [ inputValue, setInputValue ] = useState( initialState );
	//
	const handleInputChange = ( e ) => {
		//	console.log( 'handleInputChange', e.target.value );
		setInputValue( e.target.value );
	}
	//
	const handleSubmit = ( e ) => {
		e.preventDefault( );
		//	console.log( 'handleSubmit', inputValue );
		if ( inputValue.trim().length > 2 ) {
			setCategories( state => [ inputValue, ...state ] );
			setInputValue( initialState );
		}
	}
	//
	return (
		<form onSubmit={ handleSubmit }>
			<input 
				type="text"
				value={ inputValue }
				onChange={ handleInputChange }
				placeholder="..."
			/>
		</form>
	)
}
//
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}