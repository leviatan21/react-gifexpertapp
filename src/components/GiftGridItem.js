import React from 'react';
import PropTypes from 'prop-types'
//
export const GiftGridItem = ( { title, url } ) => {
	//
	return (
		<div className="card animate__animated animate__fadeIn animate__delay-2s">
			<img src={ url } alt={ title } />
			<p>{ title }</p>
		</div>
	)
}
//
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}