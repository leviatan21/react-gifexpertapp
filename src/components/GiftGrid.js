import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GiftGridItem } from './GiftGridItem';
//
export const GiftGrid = ( { category } ) => {
	//
	const { data:images, loading } = useFetchGifs( category );
	//
	return (
		<>
			<h3 className="animate__animated animate__fadeIn">{ category }</h3>
			{ loading && <p className="animate__animated animate__flash">Cargando</p> }
			<div className="card-grid">
				{ images.map( (gif) => <GiftGridItem key={ gif.id } { ...gif } /> ) }
			</div>
		</>
	)
}
