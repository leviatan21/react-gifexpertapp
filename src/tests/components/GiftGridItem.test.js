
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GiftGridItem } from '../../components/GiftGridItem';
//
describe( 'Puebas en <GiftGridItem />', () => {
	//
	const gif = {
		id:		'3ohc1292yKn6Z1saGs',
		title:	'getpunches GIF by mannyjammy',
		url:	'https://media3.giphy.com/media/3ohc1292yKn6Z1saGs/giphy.gif?cid=b1bbf17c1qvapj1ew4s6nuveegu7o50xrwvlagvr77c1reso&rid=giphy.gif'
	};
	//	console.log( gif );
	const wrapper		= shallow( <GiftGridItem title={ gif.title } url={ gif.url } /> );
	const className		= wrapper.find('div').prop('className');
	const findCard		= wrapper.find( '.card' ).length;
	const findImage		= wrapper.find( 'img' ).props();
	const findImageSrc	= findImage.src.trim();
	const findImageAlt	= findImage.alt.trim();
	const findParrafo	= wrapper.find( 'p' ).text( ).trim();
	//	
	test( 'Componente GiftGridItem', ( ) => {
		expect( wrapper ).toMatchSnapshot();
	} );
	//
	test( 'Encontrar Card', ( ) => {
		expect( findCard ).toBe( 1 );
		expect( className.includes( 'animate__fadeIn' ) ).toBe( true );
	} );
	//
	test( 'Encontrar Imagen src', ( ) => {
		expect( findImageSrc ).toBe( gif.url );
	} );
	//
	test( 'Encontrar Imagen alt', ( ) => {
		expect( findImageAlt ).toBe( gif.title );
	} );
	//	
	test( 'Encontrar Parrafo', ( ) => {
		expect( findParrafo ).toBe( gif.title );
	} );
} );
