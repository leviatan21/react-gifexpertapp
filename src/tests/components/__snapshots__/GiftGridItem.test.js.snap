// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GiftGridItem /> Componente GiftGridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="getpunches GIF by mannyjammy"
    src="https://media3.giphy.com/media/3ohc1292yKn6Z1saGs/giphy.gif?cid=b1bbf17c1qvapj1ew4s6nuveegu7o50xrwvlagvr77c1reso&rid=giphy.gif"
  />
  <p>
    getpunches GIF by mannyjammy
  </p>
</div>
`;
